src/main/java/net/openhft/chronicle/wire/YamlWire.java

```diff
@@ -383,18 +383,19 @@ private StringBuilder acquireStringBuilder() {
     @NotNull
     protected StringBuilder readField(@NotNull StringBuilder sb) {
         startEventIfTop();
-        sb.setLength(0);
         if (yt.current() == YamlToken.MAPPING_KEY) {
             yt.next();
             if (yt.current() == YamlToken.TEXT) {
-                sb.append(yt.text());
+                String text = yt.text(); // May use sb so we need to reset it after
+                sb.setLength(0);
+                sb.append(text);
                 unescape(sb, yt.blockQuote());
                 yt.next();
             } else {
                 throw new IllegalStateException(yt.toString());
             }
         } else {
-            return sb;
+            sb.setLength(0);
```

Comment:
sb.setLength() could be moved to L387 and deduplicate.

Reply:
It needs to be called after yt.text()